src/Identity/AuthenticatedIdentity.php

```diff
@@ -8,10 +8,10 @@
 
 class AuthenticatedIdentity extends Role implements IdentityInterface
 {
-    /** @var string|IdentityInterface */
+    /** @var mixed */
```

Comment:
Is it really as bad as `mixed`? 🤔

Reply:
You mean that it should be more precise? Like `string|object`? That would work for me too.

Reply:
That would certainly be better.

Do we really allow any identity though?

Reply:
I think so... The current Unit test says this:
https://github.com/laminas-api-tools/api-tools-mvc-auth/blob/1.8.x/test/Identity/AuthenticatedIdentityTest.php#L15-L21

and the second one even throws in `array` as potential type:
https://github.com/laminas-api-tools/api-tools-mvc-auth/blob/1.8.x/test/Authentication/DefaultAuthenticationListenerTest.php#L323-L331

So, following the tests, it would be `string|object|array` :thinking:  @Ocramius

Reply:
it seems to me that the `$identity` property is simply some placeholder where people are allowed to store whatever kind of identity type they want to use. So far I didn't find a relation to something else.

Reply:
Then `mixed` it is 😬

Reply:
yep, that hurts :sweat_smile:  but adding anything else will be a BC break :disappointed: